### PR TITLE
chore: get generators installation status from "@sap/bas-sdk" new api

### PR DESCRIPTION
**Pull request by EstherPerelman**
Waiting for bas-sdk release

### CODE REVIEW

File: packages/backend/src/panels/YeomanUIPanel.ts [hunk 1]
```diff
@@ -158,5 +159,6 @@ export class YeomanUIPanel extends AbstractWebviewPanel {
   public initWebviewPanel() {
     super.initWebviewPanel();
     this.webViewPanel.title = this.messages.panel_title;
+    void notifyGeneratorsInstallationProgress(this, this.webViewPanel);
```
Comment (alex-gilin) [inline packages/backend/src/panels/YeomanUIPanel.ts:162]: do you need 2 args ? `webViewPanel` is the member of YeomanUIPanel `this` instance.

File: packages/backend/src/utils/generators-installation-progress.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    // notify ui on generators are being installed in background (by sending any param - this will be detected as a message of "generators are being installed...")
+    void yeomanUIPanel.notifyGeneratorsChange(["installing generators"]);
+    // on panel disposed - stop looping in `waitForGeneratorsInstallation` function bellow
+    webviewPanel.onDidDispose(() => {
+      internal.panelDisposed = true;
+    });
+
+    await waitForGeneratorsInstallation();
+
+    if (internal.retries >= internal.MAX_RETRY) {
```
Comment (alex-gilin) [inline packages/backend/src/utils/generators-installation-progress.ts:54]: what is expected here if `yeomanUIPanel.panelDosposed === true`?

**Review by alex-gilin — CHANGES_REQUESTED** [review]

File: packages/backend/src/panels/YeomanUIPanel.ts (hunk 1 above)
Comment (EstherPerelman) [reply packages/backend/src/panels/YeomanUIPanel.ts:162]: `webViewPanel` is protected, should the code use it in a way like this: `panel['webViewPanel'] ?

File: packages/backend/src/utils/generators-installation-progress.ts (hunk 2 above)
Comment (EstherPerelman) [reply packages/backend/src/utils/generators-installation-progress.ts:54]: This will keep retrying next time - no message should be shown

**Review by alex-gilin — APPROVED** [review]

**Pull request merged**